tasks/add-fbc-contribution/add-fbc-contribution.yaml

```diff
@@ -0,0 +1,104 @@
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No valid data file was provided."
+            exit 1
+        fi
+
+        # Set the value for requestResultsFile
+        echo "$(workspaces.data.path)/ir-$(context.taskRun.uid)-result.txt" > $(results.requestResultsFile.path)
+
+        # IIB is the default internal pipeline for fbc requests
+        request=$(jq -r '.fbc.request // "iib"' ${DATA_FILE})
+
+        default_build_timeout_seconds="300"
+        iib_overwrite_from_index_credential=$(jq -r \
+          '.fbc.iibOverwriteFromIndexCredential // "iib-overwrite-fromimage-credentials"' ${DATA_FILE})
+        iib_service_config_secret=$(jq -r '.fbc.iibServiceConfigSecret // "iib-services-config"' ${DATA_FILE})
+        iib_service_account_secret=$(jq -r '.fbc.iibServiceAccountSecret // "iib-service-account"' ${DATA_FILE})
+        build_tags=$(jq -r '.fbc.buildTags // "[]"' ${DATA_FILE})
+        add_arches=$(jq -r '.fbc.addArches // "[]"' ${DATA_FILE})
+        build_timeout_seconds=$(jq -r --arg build_timeout_seconds ${default_build_timeout_seconds} \
+            '.fbc.buildTimeoutSeconds // $build_timeout_seconds' ${DATA_FILE})
+        target_index=$(jq -r '.fbc.targetIndex' ${DATA_FILE})
+        fbc_fragment=$(jq -cr '.components[0].containerImage' ${SNAPSHOT_PATH} |tr -d "\n")
+
+        # The internal-request script will create the InternalRequest and wait until it finishes to get its status
+        # If it fails (Failed, Rejected or Timed out) the script will exit and display the reason.
+        echo "Creating InternalRequest to add FBC contribution to index image:"
+        internal-request -r "${request}" \
+            -p binaryImage=$(params.binaryImage) \
+            -p fromIndex=$(params.fromIndex) \
+            -p targetIndex=${target_index} \
+            -p fbcFragment=${fbc_fragment} \
+            -p iibServiceConfigSecret=${iib_service_config_secret} \
+            -p iibServiceAccountSecret=${iib_service_account_secret} \
+            -p iibOverwriteFromIndexCredential=${iib_overwrite_from_index_credential} \
+            -p buildTimeoutSeconds=${build_timeout_seconds} \
+            -p buildTags=${build_tags} \
+            -p addArches=${add_arches} \
+            -t $(params.requestTimeout) |tee $(workspaces.data.path)/ir-$(context.taskRun.uid)-result.txt
+
+        internalRequest=$(awk 'NR==1{ print $2 }' $(workspaces.data.path)/ir-$(context.taskRun.uid)-result.txt | xargs)
+        echo "done (${internalRequest})"
+
+        # Fetching InternalRequest status and populating results
+        results=$(kubectl get internalrequest ${internalRequest} -o jsonpath='{.status.results}')
+        conditions=$(kubectl get internalrequest ${internalRequest} \
+          -o jsonpath='{.status.conditions[?(@.type=="Succeeded")]}')
+
+        jq '.genericResult' <<< "${results}" | tee $(results.fbcOptIn.path)
+        jq '.reason // "Unset"'  <<< "${conditions}" | tee $(results.requestReason.path)
```

Comment:
I have a question about this. These results are propagated to the pipeline results. I assume the release controller will then take this and save with the Release CR? Or how is it used? I'm curious, because as we discussed in another thread, this will only be returned if the internal request is successful, right? Is that ok?

Reply:
I don´t think the release controller cares about these results. They are only useful for tekton to control if the other tasks should run or not based on them. They are saved in the PipelineRun results and are also displayed in the UI.

Reply:
I see. So I guess the main goal is that the user is able to figure out what happened from the UI. And we intentionally try to make sure the internal request returns something and doesn't blow up on the internal cluster. But what about on this level? Is it actually useful to have these pipeline results? They will only be populated if it succeeds, but it's when things fail when a user is more likely to want to see more details of what happened. 

So I wonder what's the use case here - will there be something useful for the user on success? Was this part of the requirements, or did we just add this on our own? Just trying to find out if it's useful.

Reply:
In case of error, the internal-request will output what happened. The IIB internal pipeline does not populate results in case of error, so that's why we _don't care_ about it.
For the file-update and the create-advisory pipelines, the calling tasks behave differently always succeeding in the internal cluster and letting the caller to decide what to do with the errors/results.

This indeed should be changed for the FBC Release as well, but it will require changes in the internal IIB pipeline.

Reply:
OK, thanks. I agree that this can be done later.